types/MockXhrServer.d.ts

```diff
@@ -88,27 +90,60 @@ export class MockXhrServer {
    */
   addHandler(
     method: string,
-    matcher: string | RegExp | Function,
-    handler: object | Function | object[] | Function[],
-  ): MockXhrServer;
+    matcher: MockXhrServer.UrlMatcher,
+    handler: MockXhrServer.Handler
+  ): this;
 
   /**
    * Set the default request handler for requests that don't match any route.
    *
    * @param {object|Function|object[]|Function[]} handler request handler
    * @returns this
    */
-  setDefaultHandler(handler: object | Function | object[] | Function[]): MockXhrServer;
+  setDefaultHandler(handler: MockXhrServer.Handler): this;
 
   /**
    * Return 404 responses for requests that don't match any route.
    *
    * @returns this
    */
-  setDefault404(): MockXhrServer;
+  setDefault404(): this;
 
   /**
    * @returns list of requests received by the server. Entries: { method, url }
    */
-  getRequestLog(): object[];
+  getRequestLog(): MockXhrServer.RequestLog;
 }
+
+export namespace MockXhrServer {
+  type UrlMatcher =
+    ((url: string) => boolean)
+    | string
+    | RegExp
+
+  interface MockResponseHandler {
+    status: number;
+    statusText: string;
+    headers: Record<string, string>;
+    body: string;
+  }
+
+  interface MockCallbackHandler {
+    (xhr: MockXhr): void
+  }
+
+  type Handler =
+    (Partial<MockResponseHandler> | MockCallbackHandler)
```

Comment:
I'd say that it's a matter of context. If all `MockResponseHandler` members are optional regardless of its usage, then it's likely better to make the members optional, and make it an interface. 

As a rule of a thumb the mapped types shouldn't be overused, since they impose a compile-time overhead (there's no runtime impact though).